api/pkg/service/catalog/catalog.go

```diff
@@ -70,7 +65,7 @@ func (s *service) Refresh(ctx context.Context, p *catalog.RefreshPayload) (*cata
 		return nil, notFoundOrInternalError(err)
 	}
 
-	job, err := s.wq.Enqueue(user.ID, ctg.ID)
+	job, err := s.wq.Enqueue(auth.UserID(ctx), ctg.ID)
```

Comment:
perhaps UserID() should be part of auth.Service itself so that the code reads `s.UserID(ctx)` just like `Logger(ctx)` . We should keep `auth.UserID` as well if we decide to go that direction.

Reply:
We can make `UserID` method of service.. then we can do `s.User()`

Reply:
and this will be used only by API which needs authentication so the method of service would work. WDYT?

Reply:
I prefer having both :)

Reply:
okay, will address it while adding expiry time in jwt. :smile_cat: